cfme/tests/cloud_infra_common/test_cockpit_server.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.cloud.instance import Instance
+from cfme.cloud.provider import CloudProvider
+from cfme.cloud.provider.ec2 import EC2Provider
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.gce import GCEProvider
+from cfme.common.provider import CloudInfraProvider
+from cfme.infrastructure.virtual_machines import Vm
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.generators import random_vm_name
+from cfme.utils.wait import wait_for
+
+
+@pytest.fixture(scope="function")
+def new_vm(provider, request):
+    if provider.one_of(CloudProvider):
+        vm = Instance.factory(random_vm_name(context='cockpit'), provider)
+    else:
+        vm = Vm.factory(random_vm_name(context='cockpit'), provider)
```

Comment:
[It is recommended to use a maximum of 5 characters for context](https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/generators.py#L15)

Reply:
Not required since the method will auto-truncate, good spot though @vatsalparekh